### PR TITLE
Add sample Golang gRPC Client/Server programs on Occlum

### PR DESCRIPTION
**Pull request by james-dong-security**
Add sample Golang gRPC Client/Server programs on Occlum

### CODE REVIEW

**Comment by james-dong-security** [comment]
@tatetian @jessehui :
Do you mind to review the new Golang gRPC demo program with this patch?
Thanks in advance!

**Comment by dingelish** [comment]
I think it's good to go

File: demos/golang/grpc_pingpong/prepare_ping_pong.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+go mod init grpc_pingpong
+
+# build pong image
+go build pong.go
+
+# build ping image
+go build ping.go
+
+# prepare occlum images
+occlum-go build -o occlum_pong -buildmode=pie pong.go
```
Comment (liqinggd) [inline demos/golang/grpc_pingpong/prepare_ping_pong.sh:64]: In 0.15.0, PIE is the default mode for Go toolchain, so `-buildmode=pie` here is useless

File: demos/golang/grpc_pingpong/run_ping_on_occlum.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+occlum_ping="./occlum_ping"
+
+if [ ! -f $occlum_ping ];then
+    echo "Error: cannot stat file '$occlum_ping'"
+    echo "Please see README and build it using prepare_ping_pong.sh"
+    exit 1
+fi
+
+# Init Occlum Workspace
+rm -rf occlum_ping_context && mkdir occlum_ping_context
```
Comment (liqinggd) [inline demos/golang/grpc_pingpong/run_ping_on_occlum.sh]: We have renamed `occlum context` to `occlum instance`,  use `occlum_ping_instance` as the name

File: demos/golang/grpc_pingpong/run_pong_on_occlum.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+occlum_pong="./occlum_pong"
+
+if [ ! -f $occlum_pong ];then
+    echo "Error: cannot stat file '$occlum_pong'"
+    echo "Please see README and build it using prepare_pong_pong.sh"
+    exit 1
+fi
+
+# Init Occlum Workspace
+rm -rf occlum_pong_context && mkdir occlum_pong_context
```
Comment (liqinggd) [inline demos/golang/grpc_pingpong/run_pong_on_occlum.sh]: We have renamed `occlum context` to `occlum instance` in 0.15.0, use `occlum_pong_instance` as the name

File: demos/golang/grpc_pingpong/prepare_ping_pong.sh (hunk 1 above)
Comment (james-dong-security) [reply demos/golang/grpc_pingpong/prepare_ping_pong.sh]: Done

**Comment by james-dong-security** [comment]
@liqinggd 
Thanks for your comments. 
New patch is uploaded. PTAL.

**Review by liqinggd — CHANGES_REQUESTED** [review]
Could you please rebase to the master and squash all the commits into one commit?

**Comment by james-dong-security** [comment]
@liqinggd :
Actually the cloned repository got messed up. It is tricky to do rebase and squash now. I re-cloned Occlum repository, and sent out a new pull request. Would you please review following new pull request at:
    https://github.com/occlum/occlum/pull/126

**Comment by james-dong-security** [comment]
Duplicate of #126

**Comment by james-dong-security** [comment]
Duplicate of https://github.com/occlum/occlum/pull/126

**Pull request closed without merging**